consensus/src/dag/dag_store.rs

```diff
@@ -18,17 +22,41 @@ pub struct Dag {
     nodes_by_round: BTreeMap<Round, Vec<Option<Arc<CertifiedNode>>>>,
     /// Map between peer id to vector index
     author_to_index: HashMap<Author, usize>,
+    storage: Arc<dyn DAGStorage>,
 }
 
 impl Dag {
-    pub fn new(author_to_index: HashMap<Author, usize>, initial_round: Round) -> Self {
+    pub fn new(epoch_state: Arc<EpochState>, storage: Arc<dyn DAGStorage>) -> Self {
```

Comment:
what happens when the peer crashes and come back after some time? do we assume that someone outside of dag will clean up storage and populate with recent nodes before passing here?

Reply:
no, it'll recover with whatever it has before crashes (i.e. crash recovery). it's up to the sync mechanism to decide how to sync to latest state